chef/cookbooks/network/recipes/default.rb

```diff
@@ -393,6 +393,36 @@ def get_datapath_id_for_ovsbridge(bridge)
     nic.tx_offloading = node["network"]["enable_tx_offloading"] || true
   end
 
+  # Do some MTU checks/configuration for the parent of the vlan nic
+  if nic.is_a?(Nic::Vlan)
+    # 1) validate that the parent of vlan nic is not used by a network directly
+    # and the requested mtu for the parent is lower than the vlan nic mtu.
+    # That would not work and setting the mtu on the vlan would fail.
+    networks_using_parent = if_mapping.select { |net, ifaces| ifaces.include? nic.parent }.keys
+    networks_using_parent.each do |net_name|
+      net = Barclamp::Inventory.get_network_by_type(node, net_name)
+      if net.use_vlan != "true"
+        if net.mtu.to_i < ifs[nic.name]["mtu"].to_i
+          msg = "#{nic.name} wants mtu #{nic.mtu} but network #{net_name} " \
+                "using the parent nic #{nic.parent} wants a lower mtu #{net.mtu}." \
+                "This network mtu configuration is invalid."
+          Chef::Log.fatal(msg)
+          raise msg
+        end
+      end
+    end
+    # 2) set the mtu for the parent if needed
+    unless ifs[nic.parent].key? "mtu" || (ifs[nic.parent]["mtu"].to_i < ifs[nic.name]["mtu"].to_i)
```

Comment:
The move to `unless` means the second part of the test should be reverted, doesn't it?